src/network/messages.c

```diff
@@ -0,0 +1,575 @@
+/*! @file messages.c
+ * @brief defines message types and tooling for a very minimal RPC framework
+ * @note when sending messages anytime you send a new message, you must first send a
+ * single byte that indicates the size of the messsage we are sending
+ * @note this helps ensure that we can appropriately handle new requests and allocate
+ * enough memory
+ * @details the message types here are intended to be served as a building block for
+ * your own applications
+ * @details at a minimum the types here are setup to establish a secure
+ * communications channel using ECDSA keys and ECDH key agreement
+ */
+
+#include "network/messages.h"
+#include "encoding/cbor.h"
+#include "network/socket.h"
+#include "network/utils.h"
+#include "tinycbor/cbor.h"
+#include <netdb.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/socket.h>
+
+/*!
+ * @brief helper function to return a message_t object for sending
+ * @details if initiate is true, the message type is MESSAGE_HELLO_INT
+ * @details if initiate is false, the message type is MESSAGE_HELLO_FIN
+ */
+message_t *message_hello_t_to_message_t(message_hello_t *msg, bool initiate) {
```

Comment:
this and the corresponding header definition need updated comments (namely param doxygens)